src/Application/Actions/Donations/Create.php

```diff
@@ -68,6 +68,27 @@ protected function action(): Response
 
         try {
             $donation = $this->donationRepository->buildFromApiRequest($donationData);
+
+            // Optionally set these donation fields if available
```

Comment:
These setters are all doing what `buildFromApiRequest()` describes as its action, so should be in that method.

We're also missing country code.